app/src/main/java/fr/free/nrw/commons/explore/depictions/DepictsClient.java

```diff
@@ -0,0 +1,178 @@
+package fr.free.nrw.commons.explore.depictions;
+
+import androidx.annotation.Nullable;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+
+
+import java.math.BigInteger;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import fr.free.nrw.commons.Media;
+import fr.free.nrw.commons.depictions.models.Search;
+import fr.free.nrw.commons.media.MediaInterface;
+import fr.free.nrw.commons.upload.depicts.DepictsInterface;
+import fr.free.nrw.commons.upload.structure.depicts.DepictedItem;
+import fr.free.nrw.commons.utils.CommonsDateUtil;
+import io.reactivex.Observable;
+import io.reactivex.Single;
+
+@Singleton
+public class DepictsClient {
+
+    private final DepictsInterface depictsInterface;
+    private final MediaInterface mediaInterface;
+    private Map<String, Map<String, String>> continuationStore;
+
+    @Inject
+    public DepictsClient(DepictsInterface depictsInterface, MediaInterface mediaInterface) {
+        this.depictsInterface = depictsInterface;
+        this.mediaInterface = mediaInterface;
+        this.continuationStore = new HashMap<>();
+    }
+
+    /**
+     * Search for depictions using the search item
+     * @return list of depicted items
+     */
+
+    public Observable<DepictedItem> searchForDepictions(String query, int limit, int offset) {
+
+        return depictsInterface.searchForDepicts(query, String.valueOf(limit), Locale.getDefault().getLanguage(), Locale.getDefault().getLanguage(), String.valueOf(offset))
+                .flatMap(depictSearchResponse -> Observable.fromIterable(depictSearchResponse.getSearch()))
+                .map(depictSearchItem -> new DepictedItem(depictSearchItem.getLabel(), depictSearchItem.getDescription(), getImageUrl(depictSearchItem.getLabel()), false, depictSearchItem.getId()));
+    }
+
+    /**
+     *Get url for image usig image name
```

Comment:
URL
using

Reply:
sorry for the mistake, updated in the next commit